packages/transaction-controller/src/TransactionController.ts

```diff
@@ -944,6 +945,33 @@ export class TransactionController extends BaseController<
     this.update({ transactions: this.trimTransactionsForState(transactions) });
   }
 
+  /**
+   * Add params to transactionMeta.
+   *
+   * @param transactionMetaId - ID of the transaction.
+   * @param securityAlertResponse - New params to be added to tarnsaction meta.
+   */
+  updateSecurityAlertResponse(
+    transactionMetaId: string,
+    securityAlertResponse: SecurityAlertResponse,
+  ) {
+    if (!transactionMetaId || !securityAlertResponse) {
+      return;
+    }
+    const { transactions } = this.state;
+    const index = this.state.transactions.findIndex(
+      ({ id }) => id === transactionMetaId,
+    );
+    if (index >= 0) {
+      const transactionMeta = transactions[index];
+      const updatedMeta = merge(transactionMeta, { securityAlertResponse });
```

Comment:
Before updating, should we validate the status of the transaction to be updated to ensure it's in the correct stage of the lifecycle?

Typically these update methods are only applicable to `unapproved` transactions since the information is used prior to approval, does that apply here?

```
validateIfTransactionUnapproved(
  transactionMeta,
  'updateSecurityAlertResponse',
);
```

Reply:
That does not seems applicable in this case.